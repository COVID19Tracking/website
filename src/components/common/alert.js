import React from 'react'
import classnames from 'classnames'
import alertStyle from './alert.module.scss'
import alertIcon from '~images/alert/alert.svg'

const InfoboxInner = ({ header, children }) => (
  <div>
    <p className={alertStyle.header}>{header}</p>
    <div className={alertStyle.content}>{children}</div>
  </div>
)

const Alert = ({ header, children, block = false, fullSize = false }) => (
  <div
    className={classnames(
      'alert',
      alertStyle.alert,
      fullSize && alertStyle.fullSize,
      block && alertStyle.block,
    )}
  >
    <img src={alertIcon} alt="Alert icon" />
    <InfoboxInner header={header}>{children}</InfoboxInner>
  </div>
)

export default Alert
