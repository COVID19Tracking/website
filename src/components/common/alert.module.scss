.alert {
  display: inline-flex;
  padding: spacer(8);
  padding-right: spacer(32);
  margin: spacer(16) 0;
  background-color: $color-infobox-alert-light;
  color: $color-infobox-alert-dark;
  border-radius: 4px;
  @include type-size(100);

  img {
    width: toRem(16);
    margin: spacer(4) spacer(16) spacer(8) spacer(8);
  }

  p {
    margin: 0;
  }

  .header {
    font-weight: bold;
    line-height: 1em;
  }

  .content {
    margin-top: spacer(4);
  }

  &.block {
    display: flex;
  }

  &.full-size {
    height: 100%;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding-right: spacer(8);
    text-align: center;
  }
}
