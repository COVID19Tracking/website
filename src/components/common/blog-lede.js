import React from 'react'
import Byline from './byline'
import blogLedeStyles from './blog-lede.module.scss'

export default ({ headline, author, date, lede }) => (
  <div className={blogLedeStyles.lede}>
    <h1>{headline}</h1>
    <p className="blog-lede">{lede}</p>
    <Byline author={author} date={date} />
  </div>
)
