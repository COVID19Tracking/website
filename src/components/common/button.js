import React from 'react'
import { Link } from 'gatsby'

const ButtonLink = ({ big, to, children }) => (
  <Link to={to} className={`button${big ? ' button-big' : ''}`}>
    {children}
  </Link>
)

const ButtonAnchor = ({ big, href, children }) => (
  <a href={href} className={`button${big ? ' button-big' : ''}`}>
    {children}
  </a>
)

const Button = ({ big, children }) => (
  <button className={`button${big ? ' button-big' : ''}`} type="button">
    {children}
  </button>
)

export { ButtonAnchor, ButtonLink, Button }
