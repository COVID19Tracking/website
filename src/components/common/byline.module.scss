.byline {
  margin: 0.3rem 0;
  font-size: 1rem;
  a {
    font-weight: 700;
  }
  .date {
    display: inline-block;
    margin-left: 1rem;
    color: #808080;
  }
  font-size: 0.95rem;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}
