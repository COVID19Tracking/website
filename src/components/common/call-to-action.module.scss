.cta {
  color: $color-honey-700;
  .arrow {
    text-decoration: none;
    display: inline-block;
    @include margin(4, left);
  }
  &:hover {
    @include link-hover;
  }
  .new-window::after {
    @include new-window-icon($color-honey-700);
  }
}

.centered {
  display: block;
  margin: 0 auto;
  text-align: center;
  @include type-size(300);
}

.bold {
  font-weight: bold;
}

.block {
  display: block;
}

.extra-margin {
  @include margin(32, top bottom);
  display: block;
}
