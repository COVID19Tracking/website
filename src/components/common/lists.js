import React from 'react'
import listsStyles from './lists.module.scss'

const UnorderedList = ({ children }) => <ul>{children}</ul>

const OrderedList = ({ children }) => <ol>{children}</ol>

const UnstyledList = ({ children }) => (
  <ul className={`list-unstyled ${listsStyles.listUnstyled}`}>{children}</ul>
)

export { UnorderedList, OrderedList, UnstyledList }
