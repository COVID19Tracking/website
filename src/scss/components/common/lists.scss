.list-unstyled {
  list-style-type: none;
  margin: 0;
}
