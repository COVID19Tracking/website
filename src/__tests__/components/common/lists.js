import React from 'react'
import renderer from 'react-test-renderer'
import { OrderedList, UnstyledList } from '../../../components/common/lists'

const SampleItems = () => (
  <>
    <li>Item A</li>
    <li>Item B</li>
    <li>Item C</li>
  </>
)

describe('Components : Common: Lists: Ordered list', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <OrderedList>
          <SampleItems />
        </OrderedList>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})

describe('Components : Common: Lists: Unstyled list', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <UnstyledList>
          <SampleItems />
        </UnstyledList>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
