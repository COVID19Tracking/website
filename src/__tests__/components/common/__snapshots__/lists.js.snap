// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Common: Lists: Ordered list renders correctly 1`] = `
<ol>
  <li>
    Item A
  </li>
  <li>
    Item B
  </li>
  <li>
    Item C
  </li>
</ol>
`;

exports[`Components : Common: Lists: Unstyled list renders correctly 1`] = `
<ul
  className="list-unstyled listUnstyled"
>
  <li>
    Item A
  </li>
  <li>
    Item B
  </li>
  <li>
    Item C
  </li>
</ul>
`;
